Modules/_sqlite/connection.c

```diff
@@ -571,26 +570,37 @@ _pysqlite_build_py_params(sqlite3_context *context, int argc,
                     cur_py_value = Py_NewRef(Py_None);
                 }
                 break;
-            case SQLITE_BLOB:
-                buflen = sqlite3_value_bytes(cur_value);
-                cur_py_value = PyBytes_FromStringAndSize(
-                    sqlite3_value_blob(cur_value), buflen);
+            case SQLITE_BLOB: {
+                sqlite3 *db = sqlite3_context_db_handle(context);
```

Comment:
Maybe it may make sense to see what other core devs think about declaring new variables inside `case`.

Reply:
Well, there's plenty of other such cases in the code base: `grep -A1 -rE "\<case\>.*:.*{" Modules`